tests/trainer/test_trainer_error_handling.py

```diff
@@ -0,0 +1,62 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+from torch.multiprocessing.spawn import ProcessRaisedException
+
+from pytorch_lightning import Trainer
+from tests.helpers import BoringModel
+from tests.helpers.runif import RunIf
+from tests.helpers.utils import DummyException
+
+
+class TrainerStagesErrorsModel(BoringModel):
+    def on_train_start(self) -> None:
+        raise DummyException("Error during train")
+
+    def on_validation_start(self) -> None:
+        raise DummyException("Error during validation")
+
+    def on_test_start(self) -> None:
+        raise DummyException("Error during test")
+
+    def on_predict_start(self) -> None:
+        raise DummyException("Error during predict")
+
+
+@pytest.mark.parametrize("accelerator,num_processes", [(None, 1)])
+def test_error_handling_all_stages(tmpdir, accelerator, num_processes):
+    model = TrainerStagesErrorsModel()
+    trainer = Trainer(default_root_dir=tmpdir, accelerator=accelerator, num_processes=num_processes, fast_dev_run=True)
+    with pytest.raises(DummyException, match=r"Error during train"):
+        trainer.fit(model)
+    with pytest.raises(DummyException, match=r"Error during validation"):
+        trainer.validate(model)
+    with pytest.raises(DummyException, match=r"Error during test"):
+        trainer.test(model)
+    with pytest.raises(DummyException, match=r"Error during predict"):
+        trainer.predict(model, model.val_dataloader())
+
+
+@pytest.mark.parametrize("accelerator,num_processes", [pytest.param("ddp_cpu", 2, marks=RunIf(skip_windows=True))])
+def test_error_handling_all_stages(tmpdir, accelerator, num_processes):
+    model = TrainerStagesErrorsModel()
+    trainer = Trainer(default_root_dir=tmpdir, accelerator=accelerator, num_processes=num_processes, fast_dev_run=True)
+    with pytest.raises(ProcessRaisedException, match=r"Error during train"):
+        trainer.fit(model)
+    with pytest.raises(ProcessRaisedException, match=r"Error during validation"):
+        trainer.validate(model)
+    with pytest.raises(ProcessRaisedException, match=r"Error during test"):
+        trainer.test(model)
+    with pytest.raises(ProcessRaisedException, match=r"Error during predict"):
+        trainer.predict(model, model.val_dataloader())
```

Comment:
```suggestion
    with pytest.raises(ProcessRaisedException, match=r"Error during predict"):
        trainer.predict(model, model.val_dataloader(), return_predictions=False)
```
one test is failing due to this

Reply:
TIL: ProcessRaisedException isn't available in pytorch 1.6: https://pytorch.org/docs/1.6.0/_modules/torch/multiprocessing/spawn.html#spawn

for these you have to rely on regular exceptions

Reply:
>return_predictions=False
do you mind linking me to where you see the failure? I couldnt find it in CI, and locally it passes.

Reply:
Here it is:
https://github.com/PyTorchLightning/pytorch-lightning/pull/8819/checks?check_run_id=3326950706#step:6:16

Reply:
thanks @s-rog! actually I meant the error which indicates I should add "return_predictions=False"

Reply:
ah I see, my bad!

Reply:
Nvm I see it now. fixed!